src/librustc_resolve/late.rs

```diff
@@ -378,6 +378,16 @@ struct DiagnosticMetadata<'ast> {
 
     /// Only used for better errors on `let <pat>: <expr, not type>;`.
     current_let_binding: Option<(Span, Option<Span>, Option<Span>)>,
+
+    /// State used to know whether to ignore resolution errors for function bodies.
+    ///
+    /// In particular, rustdoc uses this to avoid giving errors for `cfg()` items.
+    /// In most cases this will be `None`, in which case errors will always be reported.
+    /// If it is `true`, then it will be updated when entering a nested function or trait body.
+    in_func_body: bool,
```

Comment:
Any reason this was moved out of `LateResolutionVisitor`?

Reply:
This Boolean seems to only affect diagnostics, right? Reading it again it might affect _success_ of compilation. In that case I shouldn't have moved it.

I try to segregate metadata only used for diagnostics out of the main structs to make it easier at a glance to differentiate them.

Reply:
Yeah this affects behavior, not just diagnostics. I'd prefer it be left where it was, the PR adding it was a little fragile: https://github.com/rust-lang/rust/pull/73566

Reply:
done